app/routes/tasks.py

```diff
@@ -0,0 +1,94 @@
+from datetime import datetime
+from flask import Blueprint, jsonify, request
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from app.helpers import *
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+# POST /tasks Create a new task to the db and return the task body.
+@tasks_bp.route("", methods = ["POST"])
+def post_new_task():
+
+    request_body = request.get_json()
+
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        return {"details": "Invalid data"}, 400
+    
+    new_task = Task(title=request_body["title"], description=request_body["description"], completed_at=request_body["completed_at"])
+    db.session.add(new_task)
+    db.session.commit()
+
+    return new_task.create_task_response(), 201
+
+# GET /tasks Get a list of all tasks. Handle sort requests via query params.
+@tasks_bp.route("", methods = ["GET"])
+def get_tasks():
+    sort_query = request.args.get("sort")
+    title_query = request.args.get("title")
+
+    if sort_query:
+        if sort_query == "asc":
+            tasks = Task.query.order_by(Task.title.asc())
+
+        if sort_query == "desc":
+            tasks = Task.query.order_by(Task.title.desc())
+
+    elif title_query:
+        tasks = Task.query.filter_by(title=title_query)
+
+    else:
+        tasks = Task.query.all()
+
+    return jsonify(list_of_tasks(tasks)), 200
+
+# GET /tasks/<task_id>  Get a specific task.
+@tasks_bp.route("/<task_id>", methods = ["GET"])
+@require_valid_task_id
+def get_one_task(task):
+
+    return task.create_task_response(), 200
+
+# PUT /tasks/<task_id> Update a specific task in the db and return the updated task body.
+@tasks_bp.route("/<task_id>", methods = ["PUT"])
+@require_valid_task_id
+def update_one_task(task):
+
+    request_body = request.get_json()
+    if "title" not in request_body or "description" not in request_body:
+        return {"details": "Invalid data"}, 400
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+
+    db.session.commit()
+
+    return task.create_task_response(), 200
+
+# DELETE /tasks/<task_id>  Delete a specific task and return details of deletion.
+@tasks_bp.route("/<task_id>", methods = ["DELETE"])
+@require_valid_task_id
+def delete_one_task(task):
+
+    db.session.delete(task)
+    db.session.commit()
+
+    response_body = create_details_response(f"Task {task.task_id} \"{task.title}\" successfully deleted")
+    return response_body, 200 
+
+# PATCH /tasks/<task_id>/<completion_status> Update completed at date if marked complete; change to None if marked incomplete.
+@tasks_bp.route("/<task_id>/<completion_status>", methods = ["PATCH"])
```

Comment:
We _can_ set up the route this way, but if we needed to have any other custom routes, this would match _any_ path with two parts in it (e.g., /tasks/foo/bar, /tasks/1/invalid, etc), and we would need to add logic here to handle those cases as well.

While it may seem like making to explicit routes for the complete and incomplete endpoints might be repetitive, it actually saves us from having to write the logic to pick the functionality that should be run based on the path (that's what the routing system is for). So I'd tend to simply make the two separate endpoints.